tests/codebuild/feature_tests.sh

```diff
@@ -14,10 +14,32 @@ function run_feature_canary_tests
 function run_feature_integration_tests
 {
   echo "Running feature integration tests"
+  local crd_namespace="$1"
   run_feature_canary_tests
 
-  run_test testfiles/failing-xgboost-mnist-hpo.yaml
-  run_test testfiles/failing-xgboost-mnist-trainingjob.yaml
+  run_test "${crd_namespace}" testfiles/failing-xgboost-mnist-hpo.yaml
+  run_test "${crd_namespace}" testfiles/failing-xgboost-mnist-trainingjob.yaml
+}
+
+# Creates a training job in the specified namespace
+# For this test, the job is created in a namespace that does not have the operator. 
+function run_feature_namespaced_tests
+{
+  echo "Running feature namespaced tests"
+  local crd_namespace="$1"
+  run_feature_canary_tests
+  run_test "${crd_namespace}" testfiles/xgboost-mnist-trainingjob.yaml  
```

Comment:
Verified the logic for this test by calling this method with the incorrect (default) and correct namespace to check that the test fails if job is created. 
```
agent_1  | trainingjob.sagemaker.aws.amazon.com/xgboost-mnist created
agent_1  | Verifying namespace deployment test
agent_1  | [SUCCESS] TrainingJob deployed to default namespace does not have a sagemaker job name
agent_1  | [FAILED] TrainingJob deployed to default namespace was created
```

Reply:
Why can't we just use `run_test` why do we need this extra function ?
Also I think you should add separate `test yaml` so that k8s name for the job are different.

Reply:
added test yaml.

Reply:
What do you mean by using run_test? I could have used the `wait_for_crd_status` but thats not exactly what we want here.